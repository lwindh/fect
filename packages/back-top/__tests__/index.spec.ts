/**
 *  how to test vuejs props default as  a function .
 * see : https://stackoverflow.com/questions/64434035/testing-vuejs-prop-default-that-is-anonymous-function-in-jest
 */

import BackTop from '..'
import { mount, flushPromises } from '@vue/test-utils'
import { later } from '../../../tests'

const Wrapper = (slots?: any) => {
  return {
    components: {
      [BackTop.name]: BackTop,
    },
    data: () => ({
      right: 50,
      bottom: 50,
      visibilityHeight: 10,
    }),
    template: `<div class="container" ref="container" style="height:500px;">
      <fe-backTop ref="backTopRef" :target="()=>$refs.container" :right="right" :bottom="bottom" :visibilityHeight="visibilityHeight">
        ${slots}
      </fe-backTop>
    </div>`,
  }
}

describe('BackTop', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Wrapper(), { attachTo: document.body })
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('normal props and slots should work correctly', () => {
    // fect-back-top__icon
    const wrapper = mount(Wrapper('test'), { attachTo: document.body })
    const el = wrapper.find('.fect-back-top')
    expect(el.attributes('style')).toBe('right: 50px; bottom: 50px;')
    expect(wrapper.html()).toMatchSnapshot()
    wrapper.unmount()
  })

  it('scroll event should be work', async () => {
    const wrapper = mount(Wrapper(), { attachTo: document.body })
    await wrapper.setData({
      visibilityHeight: 0,
    })

    const {
      backTopRef: { handleScroll },
    } = wrapper.vm.$refs as any
    await handleScroll(wrapper.vm.$refs.container)
    // wrapper.
    await flushPromises()
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.unmount()
  })
})
