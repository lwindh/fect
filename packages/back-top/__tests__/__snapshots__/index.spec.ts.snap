// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackTop normal props and slots should work correctly 1`] = `
<div class="container" style="height: 500px;">
  <div class="fect-back-top " style="right: 50px; bottom: 50px;">
    <!---->
  </div>
</div>
`;

exports[`BackTop scroll event should be work 1`] = `
<div class="container" style="height: 500px;">
  <div class="fect-back-top " style="right: 50px; bottom: 50px;">
    <!---->
  </div>
</div>
`;

exports[`BackTop should be render as a element 1`] = `
<div class="container" style="height: 500px;">
  <div class="fect-back-top " style="right: 50px; bottom: 50px;">
    <!---->
  </div>
</div>
`;
